pulsar/internal/commands.go

```diff
@@ -269,6 +270,117 @@ func serializeBatch(wb Buffer,
 	wb.PutUint32(checksum, checksumIdx)
 }
 
+// copy of the method serializeBatch(....) with an extension to encrypt payload
+func serializeBatchWithEncryption(wb Buffer,
```

Comment:
What are the differences between serializeBatch and serializeBatchWithEncryption? It looks like a lot of duplicate code?

Reply:
I preferred to put this in separate function instead of modifying the exiting one. Let me check if I can remove duplicate code

Reply:
reused the `serializeBatch `